clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -1003,6 +1003,9 @@ void invokeCompletedOffsetCommitCallbacks() {
         RequestFuture<Void> future =  null;
         if (!coordinatorUnknown()) {
             future = doCommitOffsetsAsync(offsets, callback);
+        } else if (offsets.isEmpty()) {
+            // No need to lookup coordinator if offsets is empty since commit of empty offsets is completed locally.
+            future = doCommitOffsetsAsync(offsets, callback);
```

Comment:
Would it make sense to do this check even before checking if the coordinator is known? Moreover, it seems that we could skip calling `doCommitOffsetsAsync` entirely by returning a completed future directly. What do you think?

```
if (offsets.isEmpty())
   return RequestFuture.voidSuccess();
```

Reply:
@dajac , we cannot return RequestFuture.voidSuccess directly here because we need hook "callbak" into the future object, which is done in doCommitOffsetsAsync.

Reply:
Oh.. I see. In this case, would it make sense to check `offsets.isEmpty()` first, then `!coordinatorUnknown()`, and the final else. That seems a bit more natural when reading it.

Reply:
Make sense. Updated.